pkg/apis/core/validation/shoot_test.go

```diff
@@ -419,6 +419,27 @@ var _ = Describe("Shoot Validation Tests", func() {
 					Expect(ValidateShootHAConfigUpdate(newShoot, shoot)).To(BeEmpty())
 				})
 			})
+
+			Context("Shoot is hibarnated", func() {
+				It("Should not allow upgrading from non-HA to HA when Spec.Hibernation.Enabled is set to `true`", func() {
+					shoot.Spec.ControlPlane = &core.ControlPlane{}
+					ValueTrue := true
+					newShoot := prepareShootForUpdate(shoot)
+					newShoot.Spec.ControlPlane = &core.ControlPlane{HighAvailability: &core.HighAvailability{FailureTolerance: core.FailureTolerance{Type: core.FailureToleranceTypeZone}}}
+					newShoot.Spec.Hibernation = &core.Hibernation{Enabled: &ValueTrue}
+					errorList := ValidateShootHAConfigUpdate(newShoot, shoot)
+					Expect(errorList).NotTo(BeEmpty())
+				})
+
+				It("Should not allow upgrading from non-HA to HA when Status.IsHibernation is set to `true`", func() {
```

Comment:
Does it make sense to add a test for the case where the cluster is not in hibernation too?

Reply:
This test is already covered [here](https://github.com/gardener/gardener/blob/master/pkg/apis/core/validation/shoot_test.go#L360) and should suffice imo